tms/models/tms_team.py

```diff
@@ -0,0 +1,120 @@
+        for team in self:
+            team.driver_count = result.get(team.id, 0)
+
+    def _compute_vehicle_count(self):
+        order_data = self.env["fleet.vehicle"].read_group(
+            [("team_id", "in", self.ids)],
+            ["team_id"],
+            ["team_id"],
+        )
+        result = {data["team_id"][0]: int(data["team_id_count"]) for data in order_data}
+        for team in self:
+            team.vehicle_count = result.get(team.id, 0)
+
+    active = fields.Boolean(default=True)
+    name = fields.Char(required=True)
+    description = fields.Text()
+    color = fields.Integer("Color Index")
+    stage_ids = fields.Many2many(
+        "tms.stage",
+        "tms_order_team_stage_rel",
+        "team_id",
+        "stage_id",
+        string="Stages",
+        default=_default_stages,
+    )
+    order_ids = fields.One2many(
+        "tms.order",
+        "team_id",
+        string="Orders",
+        domain=[("stage_id.is_closed", "=", False)],
+    )
+    order_count = fields.Integer(compute="_compute_order_count", string="Orders Count")
+    sequence = fields.Integer(default=1, help="Used to sort teams. Lower is better.")
+
+    vehicle_ids = fields.One2many("fleet.vehicle", "team_id")
+    vehicle_count = fields.Integer(
+        compute="_compute_vehicle_count", string="Vehicles Count"
+    )
+
+    driver_ids = fields.One2many(
+        "res.partner",
+        "team_id",
+        domain=[("tms_type", "=", "driver")],
+        context={"default_tms_type": "driver"},
+    )
+    driver_count = fields.Integer(
+        compute="_compute_driver_count", string="Drivers Count"
+    )
+
+    crew_ids = fields.One2many("tms.crew", "team_id")
+
+    company_id = fields.Many2one(
+        "res.company",
+        string="Company",
+        required=True,
+        index=True,
+        default=lambda self: self.env.company,
+        help="Company related to this order",
+    )
+
+    trips_todo_count = fields.Integer(
+        string="Number of Trips", compute="_compute_trips_todo_count"
+    )
+    trips_todo_count_draft = fields.Integer(
+        string="Number of Trips", compute="_compute_trips_todo_count"
+    )
+    trips_todo_count_confirmed = fields.Integer(
+        string="Number of Trips", compute="_compute_trips_todo_count"
+    )
+
+    @api.depends("order_ids.stage_id")
+    def _compute_trips_todo_count(self):
+        for team in self:
+            team.order_ids = self.env["tms.order"].search(
+                [("team_id", "=", team.id), ("stage_id.name", "!=", "Completed")]
+            )
+            data = self.env["tms.order"]._read_group(
+                [("team_id", "=", team.id), ("stage_id.name", "!=", "Completed")],
+                ["stage_id"],
+                ["__count"],
+            )
+            team.trips_todo_count = sum(count for (_, count) in data)
+            team.trips_todo_count_draft = sum(
+                count for (stage_id, count) in data if stage_id.name == "Draft"
```

Comment:
It's the same situation here. Either create an attribute in the stage or it's better not to perform this count.